tests/unit/param_types/test_notification_param.py

```diff
@@ -0,0 +1,89 @@
+import click
+import pytest
+
+from globus_cli.parsing.known_callbacks import none_to_empty_dict
+from globus_cli.parsing.param_types import NotificationParamType
+
+
+@click.command()
+@click.option("--notify", type=NotificationParamType(), callback=none_to_empty_dict)
```

Comment:
There's a check in `NotificationParamType` that can return `None`, but I don't see a test for this here.

Reply:
I think I might need to dig more deeply into how to test that behavior, for all of our parameter types.
There's no way to trigger that with "basic" usage, like these tests are doing. I _think_ that the value can be `None` when the shell completer is running. That was the case in past versions of `click`, so I've retained that habit, but I also know that the completer was largely rewritten in v8.0.

The `None` case might not even be possible anymore -- I might be operating on an old paradigm.

Reply:
I confirmed as part of my work today on how shell completion now behaves that this is an older paradigm, from `click<8.0`, and we don't receive `None` in any usage anymore. So I've removed the check, rather than adding a test for it.